src/compton/test/tCompton_NWA.cc

```diff
@@ -0,0 +1,244 @@
+  std::vector<double> etemp_evals = compton_test->get_etemp_pts();
+
+  Ensure(grp_bds.size() == 2);
+  Ensure(etemp_evals.size() == 7);
+
+  if (!soft_equiv(grp_bds[0], 3.91389000e-02)) {
+    std::cout << "Lower group bound read incorrectly!" << std::endl;
+    ITFAILS;
+  }
+  if (!soft_equiv(grp_bds[1], 5.87084000e-02)) {
+    std::cout << "Upper group bound read incorrectly!" << std::endl;
+    ITFAILS;
+  }
+
+  if (!soft_equiv(etemp_evals[0], 1.76377944e-05)) {
+    std::cout << "Etemp eval read incorrectly!" << std::endl;
+    ITFAILS;
+  }
+  if (!soft_equiv(etemp_evals[6], 6.75507064e-04)) {
+    std::cout << "Etemp eval read incorrectly!" << std::endl;
+    ITFAILS;
+  }
+
+  if (!soft_equiv(compton_test->get_min_etemp(), 1e-6)) {
+    std::cout << "Min etemp read incorrectly!" << std::endl;
+    ITFAILS;
+  }
+  if (!soft_equiv(compton_test->get_max_etemp(), 7e-4)) {
+    std::cout << "Max etemp read incorrectly!" << std::endl;
+    ITFAILS;
+  }
+
+  if (ut.numFails == 0) {
+    std::cout << "\nCorrectly read group bounds and electron temps!"
+              << std::endl;
+  }
+
+  // try "interpolating" at one of the exact eval points in the test library,
+  // and check the result:
+  std::vector<std::vector<std::vector<double>>> interp_data =
+      compton_test->interpolate(4.87227167e-04);
+
+  // Check the size of the returned data:
+  Ensure(interp_data.size() == 1);
+  Ensure(interp_data[0].size() == 1);
+  Ensure(interp_data[0][0].size() == 4);
+
+  // Check that the data is actually correct:
+  if (!soft_equiv(interp_data[0][0][0], 4.45668383e+00))
+    ITFAILS;
+  if (!soft_equiv(interp_data[0][0][1], 3.17337784e-01))
+    ITFAILS;
+  if (!soft_equiv(interp_data[0][0][2], 4.50133379e-01))
+    ITFAILS;
+  if (!soft_equiv(interp_data[0][0][3], 3.59663442e-02))
+    ITFAILS;
+
+  // get the number of xi evals in the library (we know it should be 4)
+  if (compton_test->get_num_xi() != 4)
+    ITFAILS;
+
+  if (ut.numFails == 0) {
+    std::cout << "\nCorrectly read multigroup data points!" << std::endl;
+  }
+
+  if (ut.numFails == 0) {
+    PASSMSG("Successfully linked Draco against NWA.");
+  } else {
+    FAILMSG("Did not successfully link Draco against NWA.");
+  }
+}
+
+//!  Tests the Compton_NWA mg build capability
+void compton_build_test(rtt_dsxx::UnitTest &ut) {
+  // Start the test.
+
+  std::cout << "\n------------------------------------------" << std::endl;
+  std::cout << "Test Draco call to NWA mg opacity builder" << std::endl;
+  std::cout << "------------------------------------------" << std::endl;
+
+  // open a small pointwise opacity file:
+  std::string filename = "lagrange_csk_ascii.compton";
+  std::cout << "Attempting to construct a Compton_NWA object..." << std::endl;
+
+  // make an uninitialized pointer...
+  std::shared_ptr<rtt_compton::Compton_NWA> compton_test;
+
+  // make a small fake group structure to pass in:
+  std::vector<double> test_groups(5, 0.0);
```

Comment:
Use C++11 uniform initialization instead of 5 separate assignments. Something like `test_groups = { 1,2,3,4,5};`